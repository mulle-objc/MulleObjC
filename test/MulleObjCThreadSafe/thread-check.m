#import <MulleObjC/MulleObjC.h>


//
// we only have two threads, otherwise its almost impossible without
// keeping all NSThreads around...
//
static NSThread  *NSThread_for_OSthread( mulle_thread_t arg)
{
   if( ! arg)
      return( nil);
   if( arg == MulleThreadObjectGetOSThread( [NSThread mainThread]))
      return( [NSThread mainThread]);
   return( [NSThread currentThread]);
}



int   main( void)
{
   Class      cls;
   NSThread   *obj;
   NSThread   *other;

   cls   = [NSThread class];
   obj   = [NSThread currentThread];
   other = [NSThread mulleThreadWithTarget:obj
                                  selector:@selector( mulleIsAccessibleByThread:)
                                     object:nil];
   [other mulleSetNameUTF8String:"NSThread"];

   mulle_printf( "-[%@ mulleIsThreadSafe] %btd\n",   cls, [obj mulleIsThreadSafe]);
   mulle_printf( "-[%@ threadAffinity] %@\n",        cls, NSThread_for_OSthread( _mulle_objc_object_get_thread( (struct _mulle_objc_object *) obj)));
   mulle_printf( "-[%@ mulleIsAccessible] %btd\n",   cls, [obj mulleIsAccessible]);

   mulle_printf( "-[%@ mulleIsAccessibleByThread:%@] %btd\n",
                     cls,
                     obj,
                     [obj mulleIsAccessibleByThread:obj]);

   mulle_printf( "-[%@ mulleIsAccessibleByThread:%@] %btd\n",
                     cls,
                     other,
                     [obj mulleIsAccessibleByThread:other]);

   mulle_printf( "-[%@ mulleIsAccessibleByThread:%@] %btd\n",
                     cls,
                     [NSThread mainThread],
                     [obj mulleIsAccessibleByThread:[NSThread mainThread]]);
   return( 0);
}

