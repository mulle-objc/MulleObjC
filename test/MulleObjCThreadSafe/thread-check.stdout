-[NSThread mulleIsThreadSafe] YES
-[NSThread threadAffinity] (nil)
-[NSThread mulleIsAccessible] YES
-[NSThread mulleIsAccessibleByThread:NSMainThread] YES
-[NSThread mulleIsAccessibleByThread:NSThread] YES
-[NSThread mulleIsAccessibleByThread:NSMainThread] YES
